src/redux/thunks/auth.js

```diff
@@ -1,14 +1,45 @@
-import { getAuthAPI } from '../../api/auth';
-import { setAuthData, setAuthUserPhoto } from '../actions/auth';
+import { getAuthAPI, loginAPI, logoutAPI } from '../../api/auth';
+import {
+  setAuthRequest,
+  setAuthSuccessCorrect,
+  setAuthSuccessIncorrect,
+  setAuthUserPhoto,
+  resetAuthData
+} from '../actions/auth';
 import { getProfileAPI } from '../../api/profile';
 
 export const getAuth = () => {
   return async (dispatch) => {
+    dispatch(setAuthRequest());
     const dataAuth = await getAuthAPI();
     if (dataAuth.resultCode === 0) {
-      dispatch(setAuthData(dataAuth.data));
+      dispatch(setAuthSuccessCorrect(dataAuth.data));
       const dataProfile = await getProfileAPI(dataAuth.data.id);
       dispatch(setAuthUserPhoto(dataProfile.photos.small));
+    } else if (dataAuth.resultCode === 1) {
+      dispatch(setAuthSuccessIncorrect(dataAuth.messages[0]));
+    }
+  };
+};
+
+export const login = (email, password, rememberMe, setStatus, setSubmitting) => {
+  return async (dispatch) => {
+    const data = await loginAPI(email, password, rememberMe);
+    if (data.resultCode === 0) {
+      dispatch(getAuth());
+    } else {
+      const message = data.messages[0] || 'Some error';
+      setStatus(message);
+    }
+    setSubmitting(false);
+  };
+};
+
+export const logout = () => {
+  return async (dispatch) => {
+    const data = await logoutAPI();
+    if (data.resultCode === 0) {
+      dispatch(resetAuthData(null, null, null, false, ''));
```

Comment:
For future, in case you need to pass more than 2 params into a function - use object instead, like:
```
dispatch(resetAuthData({prop1: null,prop2: null, prop3: null, prop4: false, prop5:  ''}));
```
It will be easier to read and understand =)